teps/0089-nonfalsifiable-provenance-support.md

```diff
@@ -224,24 +309,298 @@ The entrypointer image should also be able to see that this flag is set and acco
+
+The verification is done on every consumption of the TaskRun except when the TaskRun is uninitialized. When uninitialized, the 
+TektonController is not influenced by fields in the status and thus will not sign incorrect reflections of the TaskRun.
+
+The spec and annotations are not signed as there are valid interactions from other controllers or users (i.e. cancelling taskrun).
+This is fine as the controller encodes all the necessary information that we care about in the status during initialization. Editing
+the object annotations/labels or spec will not result in any unverifiable outcome of the status field.
+
+```console
+$ tkn tr describe non-falsifiable-provenance -oyaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  annotations:
+    pipeline.tekton.dev/release: 3ee99ec
+  creationTimestamp: "2022-03-04T19:10:46Z"
+  generation: 1
+  labels:
+    app.kubernetes.io/managed-by: tekton-pipelines
+  name: non-falsifiable-provenance
+  namespace: default
+  resourceVersion: "23088242"
+  uid: 548ebe99-d40b-4580-a9bc-afe80915e22e
+spec:
+  serviceAccountName: default
+  taskSpec:
+    results:
+    - description: ""
+      name: foo
+    - description: ""
+      name: bar
+    steps:
+    - image: ubuntu
+      name: non-falsifiable
+      resources: {}
+      script: |
+        #!/usr/bin/env bash
+        sleep 30
+        printf "%s" "hello" > "$(results.foo.path)"
+        printf "%s" "world" > "$(results.bar.path)"
+  timeout: 1m0s
+status:
+  annotations:
+    tekton.dev/controller-svid: |
+      -----BEGIN CERTIFICATE-----
+      MIIB7jCCAZSgAwIBAgIRAI8/08uXSn9tyv7cRN87uvgwCgYIKoZIzj0EAwIwHjEL
+      MAkGA1UEBhMCVVMxDzANBgNVBAoTBlNQSUZGRTAeFw0yMjAzMDQxODU0NTlaFw0y
+      MjAzMDQxOTU1MDlaMB0xCzAJBgNVBAYTAlVTMQ4wDAYDVQQKEwVTUElSRTBZMBMG
+      ByqGSM49AgEGCCqGSM49AwEHA0IABL+e9OjkMv+7XgMWYtrzq0ESzJi+znA/Pm8D
+      nvApAHg3/rEcNS8c5LgFFRzDfcs9fxGSSkL1JrELzoYul1Q13XejgbMwgbAwDgYD
+      VR0PAQH/BAQDAgOoMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAMBgNV
+      HRMBAf8EAjAAMB0GA1UdDgQWBBR+ma+yZfo092FKIM4F3yhEY8jgDDAfBgNVHSME
+      GDAWgBRKiCg5+YdTaQ+5gJmvt2QcDkQ6KjAxBgNVHREEKjAohiZzcGlmZmU6Ly9l
+      eGFtcGxlLm9yZy90ZWt0b24vY29udHJvbGxlcjAKBggqhkjOPQQDAgNIADBFAiEA
+      8xVWrQr8+i6yMLDm9IUjtvTbz9ofjSsWL6c/+rxmmRYCIBTiJ/HW7di3inSfxwqK
+      5DKyPrKoR8sq8Ne7flkhgbkg
+      -----END CERTIFICATE-----
+    tekton.dev/status-hash: 76692c9dcd362f8a6e4bda8ccb4c0937ad16b0d23149ae256049433192892511
+    tekton.dev/status-hash-sig: MEQCIFv2bW0k4g0Azx+qaeZjUulPD8Ma3uCUn0tXQuuR1FaEAiBHQwN4XobOXmC2nddYm04AZ74YubUyNl49/vnbnR/HcQ==
+  completionTime: "2022-03-04T19:11:22Z"
+  conditions:
+  - lastTransitionTime: "2022-03-04T19:11:22Z"
+    message: All Steps have completed executing
+    reason: Succeeded
+    status: "True"
+    type: Succeeded
+  - lastTransitionTime: "2022-03-04T19:11:22Z"
+    message: Spire verified
+    reason: checked signatures, svid, trust bundle, uri, and manifest
+    status: "True"
+    type: VERIFIED
```

Comment:
@bobcatfish Here is where the conditions would come into play like we discussed in slack. These are validating the taskrun results and this condition is set based on that. Chains would also check that this condition to ensure the results are all valid.

Reply:
ahhh okay, i missed this detail - before we merge this as implementable i think we should describe this new condition explicitly, e.g. in the `Design Details - API changes` section or similar

and as mentioned on slack i think we should try to maintain some kind of parity between how we represent that the entire taskrun status has been verified and how we represent results have been verified

Reply:
Yea let us update that section and also put in a comment the consensus we come up with.